dbms/src/DataStreams/MergingAndConvertingBlockInputStream.h

```diff
@@ -141,31 +159,39 @@ class MergingAndConvertingBlockInputStream : public IProfilingBlockInputStream
                 if (parallel_merge_data->exception)
                     std::rethrow_exception(parallel_merge_data->exception);
 
-                auto it = parallel_merge_data->ready_blocks.find(current_bucket_num);
-                if (it != parallel_merge_data->ready_blocks.end())
+
+                if (!parallel_merge_data->ready_blocks.empty())
                 {
-                    ++current_bucket_num;
                     scheduleThreadForNextBucket();
+                    for (auto & ready_block : parallel_merge_data->ready_blocks)
```

Comment:
If there is more than 1 ready_block, then it means more than 1 thread is finished, but only one thread is scheduled for next bucket, is it the expected behavior?

Reply:
> If there is more than 1 ready_block, then it means more than 1 thread is finished, but only one thread is scheduled for next bucket, is it the expected behavior?

changed, if there is n ready_block, then n threads will be scheduled